src/components/LoginButton/LoginButton.test.tsx

```diff
@@ -0,0 +1,59 @@
+import {render, screen, waitFor} from "@testing-library/react";
+import loginService from "../../services/application/loginService";
+import {BrowserRouter} from "react-router-dom";
+
+import React from "react";
+import LoginButton from "./LoginButton";
+
+const LOGIN_BUTTON_TEXT = "Login";
+
+jest.mock("../../services/application/loginService");
+
+jest.mock("react-google-login", () => {
+    return ({
+                onSuccess,
+                buttonText,
+            }: {
+        onSuccess: any;
+        buttonText: string;
+    }) => {
+        const handleClick = () => {
+            onSuccess({
+                profileObj: { name: "test name" },
+            });
+        };
+
+        return <button onClick={handleClick}>{buttonText}</button>;
+    };
+});
+
+const handleLoginRedirection = jest.fn();
+
+describe("LoginButton should", () => {
+
+    beforeEach(() => {
+        render(
+            <BrowserRouter>
+                <LoginButton onLogin={handleLoginRedirection} />
+            </BrowserRouter>
+        );
+    });
+
+    test("renders log in button", () => {
+        const button = screen.getByText(LOGIN_BUTTON_TEXT);
```

Comment:
This line can be extracted into before each

Reply:
changed!